### ISSUE TITLE
Multiple dex files define various Java classes

### ISSUE THREAD

**Issue by hicotech** [f111fei/react-native-unity-view#6]
Exporting project from Unity (android) creates file **unity-classes.jar** in **<project-folder>\android\UnityExport\libs\**

This file contains several classes which are obviously needed. For example _bitter\jnibridge\JNIBridge.class_ or _com\unity3d\player\UnityPlayer.class_. Now issue is that same file (with slightly different content of course) is also in **<project-folder>\node_modules\react-native-unity-view\android\libs\**

When running "react-native run-android" command, task ':app:transformClassesWithDexForDebug' FAILS with note "Multiple dex files define Lbitter/jnibridge/JNIBridge$a;". Removing bitter from react-native-unity-view unity-classes.jar file solves this particular duplicate but then of course there are many others.

Basically problem is that java files in src directory of both UnityExport and react-native-unity-view are importing com.unity3d.player.* from different packages (package com.reactnative.unity.view; and package com.company.project;) what is ok, but gradle doesnt like duplicated classes.

What would be solution or workaround for that?

**Comment by f111fei** [comment]
Maybe  `unity-classes.jar` in UnityExport is different with `unity-classes.jar` in react-native-unity-view.
My Unity Verison is `Unity 2017.2.0f3 (64-bit)`.  You can try copy `unity-classes.jar` to another.

**Comment by hicotech** [comment]
It is different.. react-native-unity-view has different java classes than my exported one but most of them are the same..

what I did is, that I changed path to unity-classes.jar file in build.gradle of react-native-unity-view so it uses one from UnityExport.. I can probably also combine and repackage them in gradle

```
dependencies {
    compile fileTree(dir: '../../../android/UnityExport/libs', include: ['*.jar'])
    compile 'com.facebook.react:react-native:+'
}

**Comment by hicotech** [comment]
For future reference.. 

when I run `react-native run-android` it complains about classes as described in first post..

when I run `npm run android` it doesnt complain even when there are two unity-classes.jar files with same definitions.. meaning that dependencies in react-native-unity-view build.gradle file are set to use unity-classes.jar which came with package not one from UnityExport

**Comment by f111fei** [comment]
@hicotech  thank you

**Comment by abanoubdev** [comment]
@f111fei exactly same happens while I upgraded unity from 2020.2.1 to 2020.2.2
after updating it from unity installation folder ,it works properly
